src/object.d

```diff
@@ -3323,12 +3323,6 @@ unittest
         return _dup!(const(T), T)(a);
 }
 
-/// ditto
-@property T[] dup(T:void)(scope const(T)[] a) @trusted
-{
-    import core.internal.traits : Unconst;
-    return _dup!(const(T), Unconst!T)(a);
```

Comment:
Something doesn't sit right with me, when you can `dup` a `const(void)[]` or an `immutable(void)[]` into a `void[]` without any complaint. Even in non-`@safe` code.

Yes, we have sort of fixed the `void[]` copy because it's dangerous in `@safe` code, even without violating immutability. But that's because the types could be anything, and you may be copying non-pointer data into pointer data. However, in this case, the basic types match, so that isn't the issue. That is, you won't get a dangling pointer. So it's more of a const-safety issue than memory safety.

I don't know if a unittest is appropriate here. I'm not sure what the intent of the `dup` function is at this point.